lib/src/core/networking/client.dart

```diff
@@ -538,4 +491,62 @@ abstract class OpenAINetworkingClient {
   static bool doesErrorExists(Map<String, dynamic> decodedResponseBody) {
     return decodedResponseBody[OpenAIStrings.errorFieldKey] != null;
   }
+
+  static RequestFailedException extractException(int statusCode, Map<String, dynamic> errorBody, Map<String, String> headers) {
+    final message = errorBody[OpenAIStrings.messageFieldKey];
+    final msg = message?.toString() ?? "";
+
+    if (statusCode == _rateLimitStatusCode) {
+      final remainingValue = headers[_headerRequestsRemaining];
```

Comment:
can we move the `_headerRequestsRemaining`  to the constants file of the package, just for organization.